modules/http/src/test/scala/higherkindness/mu/rpc/http/GreeterRestTests.scala

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2017-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.rpc.http
+
+import cats.effect.{IO, _}
+import fs2.Stream
+import higherkindness.mu.http.ResponseError
+import higherkindness.mu.rpc.common.RpcBaseTestSuite
+import higherkindness.mu.http.Utils._
+import io.circe.Json
+import io.circe.generic.auto._
+import io.circe.syntax._
+import monix.reactive.Observable
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.client.UnexpectedStatus
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.http4s.server.blaze._
+import org.scalatest._
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.http4s.implicits._
+import org.http4s.server.Router
+
+import scala.concurrent.duration._
+
+class GreeterRestTests
+    extends RpcBaseTestSuite
+    with GeneratorDrivenPropertyChecks
+    with BeforeAndAfter {
+
+  val Hostname = "localhost"
+  val Port     = 8080
+
+  val serviceUri: Uri = Uri.unsafeFromString(s"http://$Hostname:$Port")
+
+  val UnaryServicePrefix = "UnaryGreeter"
+  val Fs2ServicePrefix   = "Fs2Greeter"
+  val MonixServicePrefix = "MonixGreeter"
+
+  implicit val ec                   = monix.execution.Scheduler.Implicits.global
+  implicit val cs: ContextShift[IO] = IO.contextShift(ec)
+  implicit val timer: Timer[IO]     = IO.timer(ec)
+
+  implicit val unaryHandlerIO = new UnaryGreeterHandler[IO]
+  implicit val fs2HandlerIO   = new Fs2GreeterHandler[IO]
+  implicit val monixHandlerIO = new MonixGreeterHandler[IO]
+
+  //TODO: add Logger middleware
```

Comment:
Do we need to do this _TODO_?

Reply:
removed at https://github.com/higherkindness/mu/pull/552/commits/c922e0f1da911b7a61752907dc882e8522eae495 since we don't need it for the purpose of this test suite